repository/git.go

```diff
@@ -67,6 +78,67 @@ func cloneGitRepository(ctx context.Context, repo Repository, localPath string,
 	return gitRepo, nil
 }
 
+func initSubmodules(ctx context.Context, repo *git.Repository, token string, recurseSubmodules git.SubmoduleRescursivity) error {
+	if recurseSubmodules == git.NoRecurseSubmodules {
+		return nil
+	}
+	recurseSubmodules--
+
+	wt, err := repo.Worktree()
+	if err != nil {
+		return fmt.Errorf("failed to get worktree: %w", err)
+	}
+
+	subModules, err := wt.Submodules()
+	if err != nil {
+		return fmt.Errorf("failed to get submodules: %w", err)
+	}
+
+	for _, s := range subModules {
+		// Hack: rewrite Github hosted submodule SSH URLs to use HTTPS because token auth only works with that
+		// go-git does not expose a way of doing insteadOf type rewrites for submodules, so this will have to do for now
+		s.Config().URL = strings.Replace(s.Config().URL, "git@github.com:", "https://github.com/", 1)
```

Comment:
can you please use the GitHubOptions.URL here instead of hardcoding github.com, to support github enterprise?

Reply:
Do you know if github enterprise SSH clone URLs also have a custom domain name in them? i.e. `git@my-domain.tld@:org/repo.git`?

Reply:
yes, for github enterprise it's a custom domain name.